packages/manager/src/features/Domains/DownloadDNSZoneFileButton.tsx

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { LinkButton } from 'src/components/LinkButton';
+import { downloadFile } from 'src/utilities/downloadFile';
+import { getDNSZoneFile } from '@linode/api-v4/lib/domains';
+
+type Props = {
+  id: number;
+};
+
+export const DownloadDNSZoneFileButton = ({ id }: Props) => {
+  const handleClick = async () => {
+    const data = await getDNSZoneFile(id);
+    const zoneFileContent = data?.zone_file.join('\n');
+    if (zoneFileContent) {
+      downloadFile('DNSzonefile.txt', zoneFileContent);
```

Comment:
Can we have the file name be in the format `[domain].txt`? For example, `linode.com` would be  `linode.com.txt`. `DNSzonefile.txt` isn't very descriptive

```suggestion
      downloadFile(`${domain}.txt`, zoneFileContent);
```